teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -84,97 +84,399 @@ that can be shared between them, and running the second Task will be delayed by
+* Not clear how we represent the status of a Pipeline executed as a pod; i.e. do we create fake taskRuns for the
+  purposes of populating the Pipeline status (i.e. expose the status of a group of steps corresponding to one Task as
+  if they were run as a TaskRun) or simply expose it as one big TaskRun (probably not the user experience we want)
+  * We can probably find a new syntax for representing the individual tasks inside of a Pipeline executed in this way,
+    e.g. a section for each task in the pipeline, containing a subset of the overall pod status (specifically the
+    containers, results, etc., which pertain to that task)
+
+##### Supported functionality
+
+When executing an entire Pipeline as a pod, some functionality will be easily translated, some will require significant
+work, and some may not be possible in a pod. This is an initial estimate of what we can support and changes it would
+require.
+
+* Functionality that could be supported with current pod logic (e.g. by
+  [translating a Pipeline directly to a TaskRun](#pipeline-executed-as-taskrun)):
+  * Sequential tasks (specified using [`runAfter`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#using-the-runafter-parameter))
+  * [String params](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#specifying-parameters)
+  * [Array params](https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#specifying-parameters)
+  * [Workspaces](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#specifying-workspaces)
+  * [Pipeline level results](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#emitting-results-from-a-pipeline)
+  * Workspace features:
+    * [mountPaths](https://github.com/tektoncd/pipeline/blob/main/docs/workspaces.md#using-workspaces-in-tasks)
+    * [subPaths](https://github.com/tektoncd/pipeline/blob/main/docs/workspaces.md#using-workspaces-in-pipelines)
+    * [optional](https://github.com/tektoncd/pipeline/blob/main/docs/workspaces.md#optional-workspaces)
+    * [readOnly](https://github.com/tektoncd/pipeline/blob/main/docs/workspaces.md#using-workspaces-in-tasks)
+    * [isolated](https://github.com/tektoncd/pipeline/blob/main/docs/workspaces.md#isolating-workspaces-to-specific-steps-or-sidecars)
+  * Specifying Tasks in a Pipeline via [Bundles](https://github.com/tektoncd/pipeline/blob/main/docs/tekton-bundle-contracts.md)
+    (all bundles would have to be fetched before execution starts)
+  * [step templates](https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#specifying-a-step-template)
+  * [timeout](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#configuring-the-failure-timeout)
+* Functionality that could be supported with updated pod construction logic:
+  [Sidecars](https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#specifying-sidecars)
+  (may need to wrap sidecar commands such that sidecars don't start executing until their corresponding Task starts)
```

Comment:
There might be also weird case if, multiple task define similar sidecar (same name for example)

Reply:
Although I tend to see value there (in Pipeline sidecar), isn't this achievable today by starting a pod/deployment/service and removing it after (in the finally) ?

Reply:
You can achieve that today, but you have to manage it in your pipeline while it would be nice to have a more native construct for that. The other advantage of running everything in a pod for DinD is that you can rely on the socket to connect to the docker daemon and don't have to go through http/https on a different Pod port.